allenact/algorithms/onpolicy_sync/engine.py

```diff
@@ -1621,22 +1684,31 @@ def run_eval(
                 num_active_samplers=None,
                 approx_steps=None,
                 teacher_forcing=None,
-                tracking_info=None,
+                tracking_info_list=None,
                 always_enforce=True,
             )
         else:
             dist_wrapper_class = None
 
-        logging_pkg = LoggingPackage(mode=self.mode, training_steps=total_steps)
+        logging_pkg = LoggingPackage(
+            mode=self.mode,
+            training_steps=total_steps,
+            storage_uuid_to_total_experiences={},
+        )
         while self.num_active_samplers > 0:
             frames += self.num_active_samplers
-            self.collect_rollout_step(
-                rollouts, visualizer=visualizer, dist_wrapper_class=dist_wrapper_class
+            self.collect_step_across_all_task_samplers(
```

Comment:
...and changing this to ```self.collect_step_across_all_task_samplers(
                rollout_storage_uuid="_e_g_default_eval_rollout_uuid",
                uuid_to_storage={"_e_g_default_eval_rollout_uuid": rollout_storage},
                visualizer=visualizer,
                dist_wrapper_class=dist_wrapper_class,
            )```

Reply:
As mentioned above, I might miss a use case where we cannot just use the good old `RolloutBlockStorage`, but otherwise it's a minor change and we get rid of the training pipeline...

Reply:
Gotcha! One thing I'd like to be able to support is transformer-type models where we would (potentially) need to pass the whole history of observations to the agent. To be able to do this I think we'll need to have access to different types of storage than `RolloutBlockStorage`.

Reply:
I knew there had to be a reason 👍